lib/router/src/cache.rs

```diff
@@ -108,11 +115,7 @@ impl<K: Clone + Eq + Hash, V> Cache<K, V> {
                     });
                 }
 
-                Err(_e) => {
-                    return Err(CapacityExhausted {
-                        capacity: self.capacity,
-                    });
-                }
+                Err(e) => panic!("Cache.expirations DelayQueue::poll must not fail: {}", e),
```

Comment:
TIOLI: this could also be an `expect` on `self.expirations.poll`, rather than an explicit case. Not sure if it's any clearer that way.

Reply:
@hawkw I lean towards keeping the error case explicit here. It's how we seem to handle a lot of `Err(...)` cases in `match` and I think it's a little more clear this way!